astroquery/mast/utils.py

```diff
@@ -90,39 +87,115 @@ def _simple_request(url, params=None):
+                is_catalog = True
+                catalog = name
+                break
+
+    # Whether to set resolveAll to True when making the HTTP request
+    # Should be True when resolve_all = True or when object name belongs to a MAST catalog (TIC, KIC, EPIC, K2)
+    use_resolve_all = resolve_all or is_catalog
+
+    # Send request to STScI Archive Name Translation Application (SANTA)
+    params = {'name': objectname,
+              'outputFormat': 'json',
+              'resolveAll': use_resolve_all}
+    if resolver and not use_resolve_all:
+        params['resolver'] = resolver
+    response = _simple_request('http://mastresolver.stsci.edu/Santa-war/query', params)
+    response.raise_for_status()  # Raise any errors
+    result = response.json().get('resolvedCoordinate', [])
+
+    # If a resolver is specified and resolve_all is False, find and return the result for that resolver
+    if resolver and not resolve_all:
+        resolver_result = next((res for res in result if res.get('resolver') == resolver), None)
+        if not resolver_result:
+            raise ResolverError(f'Could not resolve {objectname} to a sky position using {resolver}. '
+                                'Please try another resolver or set ``resolver=None`` to use the first '
+                                'compatible resolver.')
+        resolver_coord = SkyCoord(resolver_result['ra'], resolver_result['decl'], unit='deg')
+
+        # If object belongs to a MAST catalog, check the separation between the coordinates from the
+        # resolver and the catalog
+        if is_catalog:
+            catalog_result = next((res for res in result if res.get('resolver') == catalog), None)
+            if catalog_result:
+                catalog_coord = SkyCoord(catalog_result['ra'], catalog_result['decl'], unit='deg')
+                if resolver_coord.separation(catalog_coord) > 1 * u.arcsec:
+                    # Warn user if the coordinates differ by more than 1 arcsec
+                    warnings.warn(f'Resolver {resolver} returned coordinates that differ from MAST {catalog} catalog '
+                                  'by more than 1 arcsec. ', InputWarning)
+
+        log.debug(f'Coordinates resolved using {resolver}: {resolver_coord}')
+        return resolver_coord
+
+    if not result:
+        raise ResolverError('Could not resolve {} to a sky position.'.format(objectname))
+
+    # Return results for all compatible resolvers
+    if resolve_all:
+        return {
+            res['resolver']: SkyCoord(res['ra'], res['decl'], unit='deg')
+            for res in result
+        }
+
+    # Case when resolve_all is False and no resolver is specified
+    # SANTA returns result from first compatible resolver
+    coord = SkyCoord(result[0]['ra'], result[0]['decl'], unit='deg')
+    log.debug(f'Coordinates resolved using {result[0]["resolver"]}: {coord}')
+    return coord
+
+
+def parse_input_location(coordinates=None, objectname=None, resolver=None):
```

Comment:
make all of these mandatory kwargs, too. I know it's breaking API, but it may prevent some buggy user code in the longer run